core/utils.ts

```diff
@@ -201,51 +194,49 @@ export function validate(compiledGraph: dataform.ICompiledGraph): dataform.IGrap
     if (!!action.redshift) {
       if (
         Object.keys(action.redshift).length === 0 ||
-        Object.keys(action.redshift).every(
-          key => !action.redshift[key] || !action.redshift[key].length
-        )
+        Object.values(action.redshift).every((value: string) => !value.length)
       ) {
         const message = `Missing properties in redshift config`;
         validationErrors.push(dataform.ValidationError.create({ message, actionName }));
       }
-      const redshiftConfig = [];
+
+      const validatePropertyDefined = (
+        opts: dataform.IRedshiftOptions,
+        prop: keyof dataform.IRedshiftOptions
+      ) => {
+        if (!opts[prop] || !opts[prop].length) {
+          const message = `Property "${prop}" is not defined`;
+          validationErrors.push(dataform.ValidationError.create({ message, actionName }));
+        }
+      };
+      const validatePropertiesDefined = (
+        opts: dataform.IRedshiftOptions,
+        props: Array<keyof dataform.IRedshiftOptions>
+      ) => props.forEach(prop => validatePropertyDefined(opts, prop));
+      const validatePropertyValueInValues = (
+        opts: dataform.IRedshiftOptions,
+        prop: keyof dataform.IRedshiftOptions & ("distStyle" | "sortStyle"),
```

Comment:
why do we need the distStyle | sortStyle here?

Reply:
We need to tighten the type otherwise this might be `sortKeys`, whose value is a `string[]` not `string`.